apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/ServiceCollectionExtensions.cs

```diff
@@ -35,12 +35,29 @@ public static partial class ServiceCollectionExtensions
     /// </param>
     /// <returns>The updated <see cref="IServiceCollection"/>, for method chaining.</returns>
     public static IServiceCollection AddPublisherClient(this IServiceCollection services, Action<PublisherClientBuilder> action)
+    {
+        GaxPreconditions.CheckNotNull(action, nameof(action));
+        return AddPublisherClient(services, (_, builder) => action(builder));
+    }
+
+    /// <summary>
+    /// Adds a singleton <see cref="PublisherClient"/> to the <see cref="IServiceCollection"/> as customized by the <paramref name="action"/>.
+    /// </summary>
+    /// <param name="services">
+    /// The <see cref="IServiceCollection"/> to add the singleton client to.
+    /// </param>
+    /// <param name="action">
+    /// An action delegate to invoke on the <see cref="PublisherClientBuilder"/> for configuring the <see cref="PublisherClient"/>. This is invoked before <paramref name="services"/> are used.
+    /// Must not be null and at-least <see cref="PublisherClientBuilder.TopicName"/> must be set.
+    /// </param>
+    /// <returns>The updated <see cref="IServiceCollection"/>, for method chaining.</returns>
+    public static IServiceCollection AddPublisherClient(this IServiceCollection services, Action<IServiceProvider, PublisherClientBuilder> action)
```

Comment:
This is *strictly speaking* a breaking change due to an existing call becoming ambiguous, in the following scenarios:

- Using a null literal - I'm not bothered about that, as we're validating that the action is non-null anyway (so it would only break something at compile-time that would already be broken at execution time)
- Using a method group conversion where there are two overloads: I think this is really, really unlikely, and the benefits of this outweigh the risk

Reply:
Ah, the sweet harmony of shared concerns. If you're nodding in agreement, I'm all in.